### ISSUE TITLE
Bert bi encoder ranker training with dailydialog

### ISSUE THREAD

**Issue by SirChoeg** [facebookresearch/ParlAI#1681]
I'm trying to teach the new Bert bi ranker on the daily dialog dataset. How can I get any output from the model after/during teaching? 
When I use the --display-examples true flag i get the following results:

![image](https://user-images.githubusercontent.com/11367378/56975851-f0fb7a80-6b71-11e9-92cc-6c5580dd5336.png)

There is no answer from the model.

Is there any way to try this model out?

**Comment by emilydinan** [comment]
Hi @SirChoeg -- can you let me know which command you ran? The Daily Dialog task does not have inline candidates, so you will have to provide your own set of candidates for the model to rank. Please see this section of the docs for more info: https://parl.ai/docs/tutorial_torch_ranker_agent.html#setting-candidates.

**Comment by emilydinan** [comment]
Also, in general, you can use the display_model script to display the output of your models: https://github.com/facebookresearch/ParlAI/blob/master/examples/display_model.py.

**Comment by SirChoeg** [comment]
The command I run is:
python examples/train_model.py -t dailydialog:NoStart -m bert_ranker/bi_encoder_ranker -mf models/bert_bi_daily_dialog_bs32_nostart  -bs 32 -cands batch -ecands batch -sval true  -eps 50  -tr 32 --text-truncate 32 --label-truncate 32 -opt adam -tblog true -tblog true -tbmetrics loss,nll_loss,pp --display-examples true

This should use all labels in the batch as candidates. 
Later I would like to imitate a conversation-like behavior.
I'll try the display_model script asap.

**Comment by emilydinan** [comment]
During training, when we use batch candidates with TorchRankerAgent, we don't return any text predictions: https://github.com/facebookresearch/ParlAI/blob/52364b2466124561138a23fc53fd3061118ac1b4/parlai/core/torch_ranker_agent.py#L233

So in your case, they should only display during evaluation:
https://github.com/facebookresearch/ParlAI/blob/52364b2466124561138a23fc53fd3061118ac1b4/parlai/core/torch_ranker_agent.py#L292

Part of the reason for this is speed during training. 

To imitate conversation-like behavior, use the interactive script with fixed candidates (by providing an appropriate text file with candidates): https://github.com/facebookresearch/ParlAI/blob/master/examples/interactive.py

**Comment by emilydinan** [comment]
Closing for now, feel free to re-open if you have further questions.

**Comment by SirChoeg** [comment]
I tried the interactive script with the following command:

`python examples/interactive.py -mf models/bert_bi_daily_dialog_bs32_nostart -cands fixed -ecands fixed -fixedCands options.txt`

I used a fixed cands file as you suggested (options.txt):
```
I'm fine.
Sure. I can cook.
Don't make it harder than it needs to be.
It is stable enough.
```
But after entering a message (like "How are you?")
I get an error like this:

![image](https://user-images.githubusercontent.com/11367378/57626436-2eaacb00-7596-11e9-9725-49c73f5a8e79.png)

Is my model messed up or did I miss something else?
My best guess is that I did not format my fixed candidates file appropriately.

Thanks for your quick answer.
